onnxruntime/test/qnn_ctx_gen/command_args_parser.cc

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) 2023 NVIDIA Corporation.
+// Licensed under the MIT License.
+
+#include "command_args_parser.h"
+
+#include <string.h>
+#include <iostream>
+#include <sstream>
+#include <string_view>
+#include <unordered_map>
+
+// Windows Specific
+#ifdef _WIN32
+#include "getopt.h"
+#include "windows.h"
+#else
+#include <unistd.h>
+#endif
+
+#include <core/graph/constants.h>
+#include <core/platform/path_lib.h>
+#include <core/optimizer/graph_transformer_level.h>
+
+#include "test_configuration.h"
+
+namespace onnxruntime {
+namespace qnnctxgen {
+
+/*static*/ void CommandLineParser::ShowUsage() {
+  printf(
+      "onnxruntime_qnn_ctx_gen [options...] model1_path,model2_path\n"
+      "Options:\n"
+      "\t-v: Show verbose information.\n"
+      "\t-C: Specify session configuration entries as key-value pairs: -C \"<key1>|<value1> <key2>|<value2>\" \n"
+      "\t    Refer to onnxruntime_session_options_config_keys.h for valid keys and values. \n"
+      "\t    [Example] -C \"ep.context_enable|1\" \n"
+      "\t-i: Specify QNN EP specific runtime options as key value pairs. Different runtime options available are: \n"
+      "\t    [Usage]: -i '<key1>|<value1> <key2>|<value2>'\n"
+      "\n"
+      "\t    [backend_path]: QNN backend path. e.g '/folderpath/libQnnHtp.so', '/winfolderpath/QnnHtp.dll'. default to HTP backend\n"
```

Comment:
we shouldn't even expose backend_path option right? context sharing is only for HTP

Reply:
just in case user put the htp lib in other folder so they can overwrite the default setting.

Reply:
less options the better. the way to override lib location is to use PATH or LD_LIBRARY_PATH

Reply:
Also, we can't make that assumption I think. Maybe in future, GPU could also support that? I'm thinking of changing the provider option enable_htp_weight_sharing to enable_qnn_weight_sharing.
This is a lesson learned from enable_htp_fp16_precision which is also required for GPU.